File: oop/labaratories/labaratory-work-2/Models/Pizza.cs
namespace labaratory_work_2.Models
{
    enum Dough
    {
        Thin,
        Thick
    };

    public class Pizza : Dish
    {
        public Dough Dough { get; set; } = Dough.Thick

        public Pizza() : base()
        { 
        }

        public Pizza(string name) : base(name)
        { 
        }

        public Pizza(string name, int weight, double price)
            : base(name, weight, price)
        {
        }

        public override void DisplayInfo()
        {
            return $"Pizza: {Name}, {Dough.ToString()} douph, {Weight} g, ${CalcFullPrice()}";
        }

        public override double CalcFullPrice()
        {
            return Dough == Dough.Thin ? Price * 1.3 : Price * 1.5;
        }

        public void ChangeDough()
        {
            Dough = Dough == Dough.Thin ? Dough.Thick : Dough.Thin;
        }

        public string CutInSlices()
        {
            return "Cutting the pizza into slices...";
        }
    }
}

File: oop/labaratories/labaratory-work-1/pizza.cpp
#include "pizza.h"

Dough Pizza::GetDough() const
{
	return _dough;
}

void Pizza::SetDough(Dough dough)
{
	_dough = dough;
}

void Pizza::ChangeDough()
{
	_dough = _dough == Dough::Thin ? Dough::Thick : Dough::Thin;
}

void Pizza::CutInSlices()
{
	std::cout << "Cutting the pizza into slices..." << std::endl;
}

void Pizza::DisplayInfo()
{
	std::cout << "Pizza: " << GetName() << ", "
		<< GetDough() << " dough, "
		<< GetWeight() << "g, $"
		<< std::fixed << std::setprecision(2) << CalcFullPrice()
		<< std::endl;
}

double Pizza::CalcFullPrice()
{
	return _dough == Dough::Thin ? GetPrice() * 1.3 : GetPrice() * 1.5;
}
